File: frontend/src/interfaces/Evidence.ts
// interfaces/Evidence.ts

import { type StudentInterface } from "./Student";
import { type PaymentInterface } from "./Payment";

export interface EvidenceInterface {
  EvidenceID: number;
  address: string;      // เช่น "/uploads/evidence_xxx.jpg" หรือ "uploads\xxx.jpg"
  date: string;
  note?: string;
  student_id: number;
  student_name?: string;
  student_code?: string;
  file_name?: string;
  mime_type?: string;

  StudentID: number;
  Student?: StudentInterface;

  PaymentID: number;
  Payment?: PaymentInterface;

  // CreatedAt?: Date;
  // UpdatedAt?: Date;
  // DeletedAt?: Date | null;
}

File: backend/entity/evidence.go
package entity

import (
	"time"
	"gorm.io/gorm"
)

type Evidence struct {
    gorm.Model
    File      string    `json:"file"`
    Note      string    `json:"note"`   
    Date      time.Time `json:"date"`

    // FK → Student
    StudentID uint
    Student   Student `gorm:"foreignKey:StudentID" json:"student"`

    // FK → Payment
    PaymentID uint `json:"payment_id"`
    // Payment   Payment `gorm:"foreignKey:PaymentID" json:"payment"`
}

